### PR TITLE
compute: preparation for instance controller tasks

### PR DESCRIPTION
**Pull request by teskje**
This is a collection of smaller changes I'm splitting out of the big ball of mud that is my in-progress branch for decoupling the compute controller from the coordinator (#27656). The changes are mostly concerned with the controller API, narrowing it down so that it becomes easier to support when most of the controller will live in an extra task. Other changes are minor annoyances that jumped out to me.

### Motivation

  * This PR adds a known-desirable feature.

Part of #27656 

### Tips for reviewer

I recommend looking at the commit separately.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/adapter/src/coord/sequencer/inner/explain_timestamp.rs [hunk 1]
```diff
@@ -261,7 +261,7 @@ impl Coordinator {
                         .unwrap_or_else(|| id.to_string());
                     sources.push(TimestampSource {
                         name: format!("{name} ({id}, compute)"),
-                        read_frontier: state.read_capability().elements().to_vec(),
+                        read_frontier: state.read_frontier().to_vec(),
```
Comment (teskje) [inline src/adapter/src/coord/sequencer/inner/explain_timestamp.rs]: I'm pretty sure returning the implied capability here was wrong. In EXPLAIN TIMESTAMP you either want to see the collection's actual read frontier, or the frontier at which you can still make a query, both of with are the `read_frontier()`, not the implied capability.

**Review by jkosh44 — APPROVED** [review]
Adapter changes LGTM

**Review by antiguru — APPROVED** [review]
LGTM, thanks!

**Comment by teskje** [comment]
TFTRs!

**Pull request merged**